### PR TITLE
Add netstat sample application

### PR DESCRIPTION
**Pull request by dtrugman**
An alternative suggestion to #38 and #39.

### CODE REVIEW

**Comment by dtrugman** [comment]
Thanks for the useful comments @rm5248.
The compilation was indeed broken with newer compilers. Reordering two includes fixed that.
I also agree with your comment. I was sure that I used "true" as filter in, but it turns out I didn't. Used an enum to make it explicit.

File: sample/tool_netstat.cpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{
+    LOG("=========================================================");
+    LOG("Netstat for task ID[" << task.id() << "]");
+    LOG("=========================================================");
+
+    auto inodes = task.get_fds_inodes();
+    pfs::net::net_socket_filter filter = [&inodes](const pfs::net_socket& sock){
+        return inodes.find(sock.inode) == inodes.end()
+            ? pfs::filter::action::keep
+            : pfs::filter::action::drop;
```
Comment (rm5248) [inline sample/tool_netstat.cpp:36]: The return value here is backwards, it should be:

```suggestion
        return inodes.find(sock.inode) == inodes.end()
            ? pfs::filter::action::drop
            : pfs::filter::action::keep;
```

**Comment by rm5248** [comment]
This looks pretty good to me; I think the ability to add in the custom filters is a good trade-off to make it easy to filter things from only one process while still making it be "just parse `/proc`" and simple in that regard.

File: sample/tool_netstat.cpp (hunk 1 above)
Comment (dtrugman) [reply sample/tool_netstat.cpp]: You are totally right. Thanks for noticing. Fixed.

**Comment by dtrugman** [comment]
Thank you! I'm happy you find this resolution suitable!
Will push the code, bump the version, and release it.

**Pull request merged**